azurerm/data_source_public_ip.go

```diff
@@ -67,28 +67,21 @@ func dataSourceArmPublicIPRead(d *schema.ResourceData, meta interface{}) error {
 
 	d.SetId(*resp.ID)
 
+	//ensure values are at least set to "", d.Set() is a noop on a nil
+	d.Set("fqdn", "")
+	d.Set("domain_name_label", "")
+	d.Set("ip_address", "")
+	d.Set("idle_timeout_in_minutes", 0)
+
 	if props := resp.PublicIPAddressPropertiesFormat; props != nil {
```

Comment:
in general we use `props` rather than `properties` for the local variable (mostly as it's less characters to type, not for any technical reason 🙂) - so I'd suggest this is fine as `props` here?